backend/workflow_manager/workflow_v2/workflow_helper.py

```diff
@@ -369,18 +370,25 @@ def get_status_of_async_task(
 
         Raises:
             TaskDoesNotExistError: Not found exception
+            ExecutionDoesNotExistError: If execution is not found
 
         Returns:
             ExecutionResponse: _description_
         """
-        execution = WorkflowExecution.objects.get(id=execution_id)
-
-        if not execution.task_id:
-            raise TaskDoesNotExistError()
+        try:
+            execution = WorkflowExecution.objects.get(id=execution_id)
+            if not execution.task_id:
+                raise TaskDoesNotExistError(
+                    f"No task ID found for execution: {execution_id}"
+                )
+        except WorkflowExecution.DoesNotExist:
+            raise ExecutionDoesNotExistError(
+                f"No execution found with id: {execution_id}"
+            )
```

Comment:
```suggestion
          execution = WorkflowExecution.objects.get(id=execution_id)
          if not execution.task_id:
              raise TaskDoesNotExistError(
                  f"No task ID found for execution: {execution_id}"
              )
```
@pk-zipstack ideally such validation should be done at the serializer itself when we receive an API call. Let's move it there and exit early - we'll avoid the need for such checks in inner calls

Reply:
As discussed, this does add an additional query but it can help exit early in case of failures and by adding another field to the serializer we can ideally make use of it everywhere else required